File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/NewQuestEvents.java
package tech.zolhungaj.amqapi.servercommands.globalstate;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;

import java.util.List;

@CommandType("new quest events")
public record NewQuestEvents(
        @Json(name = "events")
        List<EventsItem> quests
){
    public record EventsItem(
            @Json(name = "questEvent")
            QuestEventState questEventState,
            Data data
    ){}

    public enum QuestEventState{
        @Json(name = "1")
        STATE_UPDATE()
    }
    public record Data(QuestState newState){}

    public record QuestState(
            int questId,
            int ticketReward,
            int noteReward,
            @Json(name = "state") int currentValue,
            @Json(name = "targetState") int targetValue,
            @Json(name = "weekSlot") int dayOfWeek,
            @Json(name = "name") String questName,
            String description
            ) {}
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/NewQuestEvents.kt
package tech.zolhungaj.amqapi.servercommands.globalstate

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.QuestEventItem

@JvmRecord
@CommandType("new quest events")
data class NewQuestEvents(
    @Json(name = "events") val quests: List<QuestEventItem>
)

